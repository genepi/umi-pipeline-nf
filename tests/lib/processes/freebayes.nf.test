nextflow_process {

    name "Test Process FREEBAYES"
    script "lib/processes/freebayes.nf"
    process "FREEBAYES"

    test("Should run without failures") {

        when {
            params {
                variant_calling = true 
                variant_caller = "freebayes"
            }
            process {
                """
                input[0] = Channel.of(["barcode03", "final", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"])
                input[1] = Channel.of("final")
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("Should output three files") {

        when {
            params {
                debug = true
                variant_calling = true 
                variant_caller = "freebayes"
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", "final", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"],
                    ["barcode04", "final", "$baseDir/tests/input/variant_calling/barcode04/final.bam", "$baseDir/tests/input/variant_calling/barcode04/final.bam.bai"],
                    ["barcode05", "final", "$baseDir/tests/input/variant_calling/barcode05/final.bam", "$baseDir/tests/input/variant_calling/barcode05/final.bam.bai"]
                    )
                input[1] = Channel.of("final", "final", "final")
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}
