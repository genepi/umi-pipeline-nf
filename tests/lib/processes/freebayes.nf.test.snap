{
    "Should output three files": {
        "content": [
            {
                "0": [
                    "final.vcf:md5,88efc7556e75a4896acb25af4fd891ea",
                    "final.vcf:md5,8d2f225076db263df46ea74c3444d8a1",
                    "final.vcf:md5,c1ef037ccb4f791326fda68ef02ab497"
                ],
                "variants": [
                    "final.vcf:md5,88efc7556e75a4896acb25af4fd891ea",
                    "final.vcf:md5,8d2f225076db263df46ea74c3444d8a1",
                    "final.vcf:md5,c1ef037ccb4f791326fda68ef02ab497"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-11T12:43:44.072561"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "final.vcf:md5,c1ef037ccb4f791326fda68ef02ab497"
                ],
                "variants": [
                    "final.vcf:md5,c1ef037ccb4f791326fda68ef02ab497"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-11T12:43:32.975611"
    }
}